File: GFG/Sequence Fun/Sequence Fun.java
class Solution
{
    public int NthTerm(int n)
    {
        // code here
        long prev=2L;
        long mod=1000000007;
        for(int i=2;i<=n;i++){
            long curr=(prev%mod * i%mod)+1;
            prev=curr;
        }
        return (int)prev;
   

File: GFG/Sequence Fun/Sequence Fun.cpp
class Solution {
	public:
		int NthTerm(int n){
		    // Code  here
		    
		    long mod = 1000000007;
		    
		    long ans = 1;
		    
		    for(int i=1; i<=n; i++){
		        
		        ans = (ans*i)% mod + 1;
		        
		    }
		    
		    
		    return ans;
		}

};
